real loop indexes used in mbpf

GDIT is testing a static analysis tool called Codee (https://www.codee.com/).  Codee flagged some issues with mgbf.

codee: error: failed to analyze '/lfs/h1/hpc/support/USER/Projects/GSI/develop/src/mgbf/mg_input.f90'

error: Semantic errors in /lfs/h1/hpc/support/USER/Projects/GSI/develop/src/mgbf/mg_input.f90
/lfs/h1/hpc/support/USER/Projects/GSI/develop/src/mgbf/mg_input.f90:67:12: error: Must have INTEGER type, but is REAL(4)
           V(n,m)=ampl*(mg*imax0+ng)
             ^
/lfs/h1/hpc/support/USER/Projects/GSI/develop/src/mgbf/mg_input.f90:67:14: error: Must have INTEGER type, but is REAL(4)
           V(n,m)=ampl*(mg*imax0+ng)
               ^
/lfs/h1/hpc/support/USER/Projects/GSI/develop/src/mgbf/mg_input.f90:145:12: error: Must have INTEGER type, but is REAL(4)
           V(l,n,m)=ampl*(mg*imax0+ng) +(l-1)*incrm
             ^
/lfs/h1/hpc/support/USER/Projects/GSI/develop/src/mgbf/mg_input.f90:145:14: error: Must have INTEGER type, but is REAL(4)
           V(l,n,m)=ampl*(mg*imax0+ng) +(l-1)*incrm
               ^
/lfs/h1/hpc/support/USER/Projects/GSI/develop/src/mgbf/mg_input.f90:145:16: error: Must have INTEGER type, but is REAL(4)
           V(l,n,m)=ampl*(mg*imax0+ng) +(l-1)*incrm

The l,n,m variables are loop indexes but they are declared.
real(i_kind):: ng,mg,L,m,n

--- a/src/mgbf/mg_input.f90
+++ b/src/mgbf/mg_input.f90
@@ -57,7 +57,7 @@ integer(i_kind),intent(in):: imin,jmin
 integer(i_kind),intent(in):: imax0
 integer(i_kind),intent(in):: ampl
 real(r_kind),dimension(imin:imax,jmin:jmax),intent(out):: V
-real(i_kind):: ng,mg,L,m,n
+integer(i_kind):: ng,mg,L,m,n
 !-----------------------------------------------------------------------
 
      do m=imin,jmax
@@ -134,7 +134,7 @@ integer(i_kind),intent(in):: imax,jmax,lmax
 integer(i_kind),intent(in):: imax0
 integer(i_kind),intent(in):: ampl,incrm
 real(r_kind),dimension(lmin:lmax,imin:imax,jmin:jmax),intent(out):: V
-real(i_kind):: ng,mg,L,m,n
+integer(i_kind):: ng,mg,L,m,n
 !-----------------------------------------------------------------------
 
    do l=lmin,lmax

